app/models/movie_search.rb

```diff
@@ -13,30 +14,27 @@ def initialize(title, in_db, tmdb_id, release_date, vote_average, overview, back
     @lists = lists
   end
 
-  attr_accessor :title, :in_db, :tmdb_id, :release_date, :vote_average, :overview, :backdrop_path, :poster_path, :tags, :lists
-
-  def self.parse_results(json)
-    @movies = []
-    json.each do |result|
-      @tmdb_id = result[:id]
-      @in_db = Movie.exists?(tmdb_id: @tmdb_id)
-        if @in_db
-          @movies << Movie.find_by(tmdb_id: @tmdb_id)
-        else
-          @title = result[:title]
-          @release_date = result[:release_date]
-          @vote_average = result[:vote_average]&.round(1)
-          @overview = result[:overview]
-          @backdrop_path = result[:backdrop_path]
-          @poster_path = result[:poster_path]
-          @tags = nil
-          @lists = nil
+  def self.parse_results(results)
+    results.map do |result|
+      tmdb_id = result[:id]
+      existing_movie = Movie.find_by(tmdb_id: @tmdb_id)
 
-          @movie = MovieSearch.new(@title, @in_db, @tmdb_id, @release_date, @vote_average, @overview, @backdrop_path, @poster_path, @tags, @lists)
-          @movies << @movie
-        end
-    end #loop
-    @movies
+      if existing_movie.present?
+        existing_movie
+      else
+        new(
+          title: result[:title],
+          in_db: false,
+          tmdb_id: tmdb_id,
+          release_date: result[:release_date],
+          vote_average: result[:vote_average]&.round(1),
+          overview: result[:overview],
+          backdrop_path: result[:backdrop_path],
+          poster_path: result[:poster_path],
+          tags: nil,
+          lists: nil
```

Comment:
I'm wondering about these `tags` and `lists` and if we need these attributes at all. This object is not persisted and adding that data would require persistence... 🤔